### ISSUE TITLE
Inconsistency in "do something else simultaneously"

### ISSUE THREAD

**Issue by rayanth** [bitburner-official/bitburner-src#710]
When working for a faction or corp, if you click "do something else simultaneously", it returns to the faction details or corp location window

When working out at a gym, or studying at a university, if you click "do something else simultaneously", it returns to the City screen for the city the player is in.

In most cases, at least for myself, when I am working out at the gym, I am trying to target specific Str/Def/Dex/Agi scores, e.g. for faction invites or to get infiltration difficulty down, so when I am working on one, I wait for it to hit the target number then immediately switch to the next training type. For this reason, it would be preferable if "do something else simultaneously" returned me to the Gym interface, rather than the City - it is fewer clicks to get to the next training type that I wish to do. 

I don't think this would be quite as important for the University studying, because I don't find myself switching between different courses, but for consistency's sake, it probably should also return the player to the University screen rather than the City screen

editing to clarify a little about the "inconsistency" claim: 
When working for a Faction or Corp, the "Do something else simultaneously" button essentially brings you back to the screen that you had to be on to get to that active task, but when studying or working out, it brings you back to the city screen for the city you were in - I am proposing that for studying/working out, it should also bring you back to the screen you had to be on to get to that task - either that gym's interface, or that university's interface

**Comment by Snarling** [comment]
I believe the reason for the inconsistency is:

Company or Faction work: There are things to do on the previous screen that don't completely invalidate your active work.
Gym or university: There is nothing to do at these locations other than start an entirely new training.

As far as training-swaps, if you want to train strength until X and then start training defense, it's not ideal to start strength training and then go to "do something else" until you get to the strength level you wanted, because there is a penalty on training speed for being unfocused. So you should stay focused until you get to the desired strength level, then there is little impact in whether you cancel out of the active work (to start a new one) or "do something else simultaneously" just to go cancel out of the current work by starting a new one.

Basically, if you're wanting to navigate back to the university/gym screen, you're not actually trying to "do something else simultaneously," you're trying to do switch to a completely different active work task.

**Comment by HiEv** [comment]
> there is a penalty on training speed for being unfocused.

Isn't that only true if you don't have the Neuroreceptor Management Implant augment?

And even if that weren't true, your argument there is still kind of irrelevant, since I can certainly think of cases where I'd want to be doing something else and getting the improvements from gym training at the same time, even if the training rate is lower.

Personally, I'd rather see buttons work consistently, as Rayanth describes.

**Comment by myCatsName** [comment]
hey @rayanth , I'm just curious, do you know any way to write a script to do the type of gym work you want? What if you could use a script to go to the gym and workout, and not have to click for yourself?

**Comment by HiEv** [comment]
> do you know any way to write a script to do the type of gym work you want? What if you could use a script to go to the gym and workout, and not have to click for yourself?

That question is off-topic here, but if you don't mind a bit of a spoiler, [it is possible after unlocking a certain aspect of the game](https://github.com/bitburner-official/bitburner-src/blob/dev/markdown/bitburner.singularity.md).

**Comment by Snarling** [comment]
> And even if that weren't true, your argument there is still kind of irrelevant, since I can certainly think of cases where I'd want to be doing something else and getting the improvements from gym training at the same time, even if the training rate is lower.

Yes, and you can already do that, it just doesn't pop you back to the gym or university screen. My point is that you can't do something else simultaneously at the gym, you can only start a replacement task there, so that is why "do something else simultaneously" doesn't currently pop you back to the originating location.

But I have no problem with changing it for consistency. I was just explaining why there is currently a difference between university/gym (where the originating location has nothing else to do simultaneously) and the other work tasks.

**Comment by myCatsName** [comment]
> > do you know any way to write a script to do the type of gym work you want? What if you could use a script to go to the gym and workout, and not have to click for yourself?
> 
> That question is off-topic here, but if you don't mind a bit of a spoiler, [it is possible after unlocking a certain aspect of the game](https://github.com/bitburner-official/bitburner-src/blob/dev/markdown/bitburner.singularity.md).

Exactly. I know that exists, and I could write some other ways to do the same thing. But if OP @rayanth  does _not_ know the game actually gives you those "powers" later, it could be a sign of not communicating expectations to the player, letting them know something about the "scope" and possibilities in the game. I'm curious about new-player experience, since in Discord we also see new players for example, doing soft resets before finishing fl1ght.exe, because they feel they are "moving too fast" not ready. So maybe in game communications (direct or NPC based) can do more foreshadowing, about API upgrades, or the "cyclic" nature of play.

**Comment by rayanth** [comment]
I am aware that you can script the gym/university/etc. I primarily use the Alain Bryden scripts, which are orders of magnitude more complex than anything I can currently write myself, and those scripts prefer to use other methods to get the stats that could normally be gleaned from gym work. Due to their complexity, I'm not sure where I could add that into them myself, and if I were to try to do it as my own script it would conflict with those ones, so I'd have to turn them off in the meantime... it's easier to just do the gym work manually. 

I can understand the argument that the gym screen doesn't present the opportunity to actually do something else simultaneously... at the risk of rabbit-holing, if the intent is to prevent the user from seeing other things they can NOT do simultaneously, then it could be argued that locations that do not provide simultaneous work should be disabled when the user clicks that button, and there's dozens of arguments for why that's unwise.

I also considered suggesting the addition of a feature that actually LETS you do multiple selected workouts at the gym simultaneously, or perhaps just one new button to do all four simultaneously, but reducing the exp rate appropriately - i.e., if doing Str and Agi at the same time, each would grow at 1/2 their normal exp rate. (In the real world, most people rotate their workouts at the gym to work multiple different muscle groups and/or skills)
